src/library/starrating.cpp

```diff
@@ -30,20 +31,21 @@ StarRating::StarRating(
 }
 
 QSize StarRating::sizeHint() const {
-    return PaintingScaleFactor * QSize(m_maxStarCount, 1);
+    return m_scaleFactor * getbaseSize();
 }
 
-void StarRating::paint(QPainter *painter, const QRect &rect) const {
+void StarRating::paint(QPainter* painter, const QRect& rect, int height) {
+    m_scaleFactor = height > 0 ? height : kDefaultPaintingScaleFactor;
     // Assume the painter is configured with the right brush.
     painter->setRenderHint(QPainter::Antialiasing, true);
     painter->setPen(Qt::NoPen);
 
-    int yOffset = (rect.height() - PaintingScaleFactor) / 2;
+    int yOffset = (rect.height() - m_scaleFactor) / 2;
```

Comment:
subtracting a factor form a value sound wrong. If this is relay correct here, please add am comment.